torch/csrc/jit/profiling_record.cpp

```diff
@@ -18,7 +18,67 @@ ProfileOp* ProfilingRecord::createProfileNode(
   return pn;
 }
 
-void ProfilingRecord::instrumentBlock(Block* block) {
+static void unprofileGraphInputs(const std::shared_ptr<Graph> &graph) {
+  for (auto i : graph->inputs()) {
+    if (i->type()->isSubtypeOf(TensorType::get())) {
+      i->setType(unshapedType(i->type()));
+    }
+  }
+}
+
+static void unprofileBlock(Block *block) {
+  for (auto it = block->nodes().begin(); it != block->nodes().end(); ++it) {
```

Comment:
Would be nice not to recompute the `end` iterator every time: https://llvm.org/docs/CodingStandards.html#don-t-evaluate-end-every-time-through-a-loop

Reply:
will fix